### ISSUE TITLE
Update how ChAs are found for the meet your ChA section

### ISSUE THREAD

**Issue by viviancan** [Iridescent-CM/technovation-app#5107]
The `find_chapter_ambssador_by` method needs to be update to find accounts with chapter ambassador profiles that have a chapter id. the chapter should also have an affiliation agreement on file

**Comment by viviancan** [comment]
Refs #4814

**Comment by dboyer** [comment]
Update: this ticket will cover hooking up the Meet your ChA section to show the chapter details for the chapter that the participant selected.

**Comment by viviancan** [comment]
This is on Preview

**Comment by viviancan** [comment]
@dboyer This should pull the chapter/cha information of the selected chapter for students

**Comment by dboyer** [comment]
Tested on preview and confirmed:

- Student who was already associated with Kenya chapter sees the Kenya info in the Meet your ChA section
- Student associated with Missoula chapter sees the Missoula info in the Meet your ChA section
- Logged in as a student associated with a Nigeria chapter and saw the correct info. As an admin, I assigned the student to a different Nigeria chapter. Logged in as the student and saw the updated chapter info.
- Logged in an unaffiliated student and see the general "not affiliated with a chapter" text
- Registered as a new student with no chapter in my country and saw the "not affiliated with a chapter" text
- Registered as a new student and selected a chapter and saw the correct info in the Meet your ChA section
